### PR TITLE
feat: [M3-6816] - Add AGLB Create Route Drawer

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Implements the AGLB create route drawer.

## Preview 📷
| Drawer      | Error |
| ----------- | ----------- |
|Screenshot 2023-10-17 at 1 24 39 PM|Screenshot 2023-10-17 at 1 24 49 PM |

### Prerequisites
- Checkout this PR
- Turn on the MSW (bottom left dev tools)

### Reproduction steps
- Go to http://localhost:3000/loadbalancers/0/routes
- Click `Create Route`

### Verification steps 
- Observe drawer opens with label input field and protocol options.
- Observe when click `Create Route` _without_ a label, an error is surfaced.
- Run cypress tests `yarn cy:run -s cypress/e2e/core/loadBalancers/load-balancer-routes.spec.ts`

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/CreateRouteDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          label="Route Label"
+          name="label"
+          onChange={formik.handleChange}
+          value={formik.values.label}
+        />
+        <RadioGroup
+          onChange={(_, value) => formik.setFieldValue('protocol', value)}
+          value={formik.values.protocol}
+        >
+          <FormLabel sx={{ marginTop: theme.spacing(1) }}>Protocol</FormLabel>
```
Comment (bnussman-akamai) [inline packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/CreateRouteDrawer.tsx:75]: We could remove 
```ts
const theme = useTheme();
```
if we do
```suggestion
          <FormLabel sx={(theme) => ({ marginTop: theme.spacing(1) })}>
            Protocol
          </FormLabel>
```

**Review by bnussman-akamai — APPROVED** [review]
Looks great! 

Optional change but we might want to make this form have more dynamic error handling to stick to our UX best practices.


I started doing this in the `RuleDrawer.tsx` for example. Our patterns around dynamic error handling are a bit all over the place because formik makes it difficult so feel free to explore the options.

File: packages/validation/src/loadbalancers.schema.ts [hunk 2]
```diff
@@ -63,6 +63,11 @@ export const CreateServiceTargetSchema = object({
   healthcheck: HealthCheckSchema,
 });
 
+export const CreateRouteSchema = object({
```
Comment (coliu-akamai) [inline packages/validation/src/loadbalancers.schema.ts:66]: should we add a changeset for the Validation package here?

File: packages/manager/cypress/support/intercepts/load-balancers.ts [hunk 3]
```diff
@@ -103,6 +104,7 @@ export const mockUploadLoadBalancerCertificate = (
  *
  * @returns Cypress chainable.
  */
+// TODO: We should probably remove this mock and use "mockGetLoadBalancerServiceTargets" below.
```
Comment (coliu-akamai) [inline packages/manager/cypress/support/intercepts/load-balancers.ts:107]: if this is an AGLB specific todo, could be worth switching to // TODO: AGLB to be more consistent with the other AGLB todo comments

```suggestion
// TODO: AGLB - We should probably remove this mock and use "mockGetLoadBalancerServiceTargets" below.
```

**Review by coliu-akamai — COMMENTED** [review]
Looks awesome! 🎉 
✅ confirmed tests
✅ confirmed drawer appears
✅ confirmed error message appears if no label put in

Had a couple of questions, both in comments and below:

- the mockup has the text 'Routes Creation info' beneath the drawer title -- do we still plan on including this text or are there more up to date mockups I can cross reference?
- For vpc error handling, we had a few tickets (see M3-7170) where we ensured errors did not persist after making changes to a field. Would we want to include that behavior here/with other AGLB forms? This is what currently happens - typing in the label field doesn't get rid of the error:
image

**Comment by jaalah-akamai** [comment]
@coliu-akamai The new way of error handling should clear the error now.

> the mockup has the text 'Routes Creation info' beneath the drawer title -- do we still plan on including this text or are there more up to date mockups I can cross reference?

@bnussman-akamai I omitted the text because it didn't seem legit - thoughts?

**Comment by bnussman-akamai** [comment]
@jaalah-akamai You are correct, all copy isn't final so it's fine if we omit it. We will be coming back later to update and add copy anyway

**Review by coliu-akamai — APPROVED** [review]
Thanks for confirming, thanks Jaalah & Banks!
Confirmed error gets cleared 🎉

**Review by bnussman-akamai — APPROVED** [review]

**Pull request merged**